test/suites/tuned/workload-partitioning.robot

```diff
@@ -0,0 +1,258 @@
+*** Settings ***
+Documentation       Tests for Workload partitioning
+
+Resource            ../../resources/microshift-config.resource
+Resource            ../../resources/common.resource
+Resource            ../../resources/systemd.resource
+Resource            ../../resources/microshift-process.resource
+Resource            ../../resources/microshift-network.resource
+
+Suite Setup         Setup Suite With Namespace
+Suite Teardown      Teardown Suite
+
+
+*** Variables ***
+${MANAGEMENT_CPU}                       0
+${SYSTEMD_CRIO_DROPIN}                  /etc/systemd/system/crio.service.d/microshift-cpuaffinity.conf
+${SYSTEMD_MICROSHIFT_DROPIN}            /etc/systemd/system/microshift.service.d/microshift-cpuaffinity.conf
+${SYSTEMD_OVS_DROPIN}                   /etc/systemd/system/ovs-vswitchd.service.d/microshift-cpuaffinity.conf
+${SYSTEMD_OVSDB_DROPIN}                 /etc/systemd/system/ovsdb-server.service.d/microshift-cpuaffinity.conf
+${CRIO_CONFIG_DROPIN}                   /etc/crio/crio.conf.d/20-microshift-wp.conf
+${KUBELET_CPU_STATE_FILE}               /var/lib/kubelet/cpu_manager_state
+${KUBELET_WORKLOAD_PINNING_CONFIG}      /etc/kubernetes/openshift-workload-pinning
+
+
+*** Test Cases ***
+Control Plane Pods Must Be Annotated
+    [Documentation]    Verify that all the Control Plane pods are properly annotated.
+    All Pods Should Be Annotated As Management
+
+Workload Partitioning Should Work
+    [Documentation]    Verify that all the Control Plane pods are properly annotated.
+    [Setup]    Setup For Workload Partitioning
+    Microshift Services Should Be Running On Reserved CPU    ${MANAGEMENT_CPU}
+    All Pods Should Run On Reserved CPU    ${MANAGEMENT_CPU}
+    Create Hello MicroShift Pod
+    All Pods Should Run On Reserved CPU    ${MANAGEMENT_CPU}    true
+    [Teardown]    Teardown For Workload Partitioning
+
+
+*** Keywords ***
+Setup For Workload Partitioning
+    [Documentation]    Setup for Workload Partitioning
+    Save Default MicroShift Config
+    Configure Kubelet For Workload Partitioning    ${MANAGEMENT_CPU}
+    Configure CRIO For Workload Partitioning    ${MANAGEMENT_CPU}
+    Configure CPUAffinity In Systemd    ${MANAGEMENT_CPU}    ${SYSTEMD_CRIO_DROPIN}
+    Configure CPUAffinity In Systemd    ${MANAGEMENT_CPU}    ${SYSTEMD_MICROSHIFT_DROPIN}
+    Configure CPUAffinity In Systemd    ${MANAGEMENT_CPU}    ${SYSTEMD_OVS_DROPIN}
+    Configure CPUAffinity In Systemd    ${MANAGEMENT_CPU}    ${SYSTEMD_OVSDB_DROPIN}
```

Comment:
What do you think about doing all this in a blueprint? So when the system is up then everything is already configured?

Reply:
yes, it makes sense , but i will have to change the `reservedSystemCPUs: 0-1` in your  blueprint  to `reservedSystemCPUs: 0` , since this test  dont work very well with range of CPUs `0-1`, wdyt?

Reply:
we agreed off line that we can do this optimization later